kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterScheduler.scala

```diff
@@ -70,121 +72,50 @@ private[spark] class KubernetesClusterScheduler(conf: SparkConf)
     val kubernetesHost = "k8s://" + client.getMasterUrl().getHost()
     logInfo("Using as kubernetes-master: " + kubernetesHost.toString())
 
-    var annotationMap = Map("pod.beta.kubernetes.io/init-containers" -> raw"""[
-                {
-                    "name": "client-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/client.jar", "$clientJarUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "distro-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/spark.tgz", "$sparkDistUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "setup",
-                    "image": "$sparkDriverImage",
-                    "command": ["./install.sh"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        },
-                        {
-                            "name": "opt",
-                            "mountPath": "/opt"
-                        }
-                    ]
-                }
-            ]""")
-
-
     val labelMap = Map("type" -> "spark-driver")
     val pod = new PodBuilder()
       .withNewMetadata()
       .withLabels(labelMap.asJava)
       .withName(driverName)
-      .withAnnotations(annotationMap.asJava)
       .endMetadata()
       .withNewSpec()
       .withRestartPolicy("OnFailure")
+      .withServiceAccount(serviceAccountName)
       .addNewContainer()
       .withName("spark-driver")
       .withImage(sparkDriverImage)
       .withImagePullPolicy("Always")
-      .withCommand("/opt/spark/bin/spark-submit")
-      .withArgs(s"--class=${args.userClass}",
+      .withCommand(s"/opt/driver.sh")
+      .withArgs(s"$clientJarUri",
+                s"--class=${args.userClass}",
                 s"--master=$kubernetesHost",
                 s"--executor-memory=${driverDescription.mem}",
                 s"--conf=spark.executor.jar=$clientJarUri",
                 s"--conf=spark.executor.instances=$instances",
-                s"--conf=spark.kubernetes.namespace=${getNamespace()}",
+                s"--conf=spark.kubernetes.namespace=$nameSpace",
                 s"--conf=spark.kubernetes.driver.image=$sparkDriverImage",
-                s"--conf=spark.kubernetes.distribution.uri=$sparkDistUri",
-                "/work-dir/client.jar",
+                "/opt/spark/kubernetes/client.jar",
```

Comment:
Good point. I think that is a healthy assumption to make. I believe this was a first attempt with room for improvement. Right now, this pr works by doing a curl on the provided url for the jar and saving it has `client.jar` and this is done in the `driver.sh` script.  I believe Taking a look at the scripts:  [driver.sh](https://github.com/erikerlandson/openshift-spark/blob/k8s-native/scripts/spark/added/driver.sh) and [executor.sh](https://github.com/erikerlandson/openshift-spark/blob/k8s-native/scripts/spark/added/executor.sh) from @erikerlandson repo will give you more clearer context.